src/stackmap_checker/call_stack_state.c

```diff
@@ -235,3 +402,22 @@ frame_t* alloc_empty_frames(size_t num_frames)
     }
     return frames;
 }
+
+
+void free_frames(frame_t *frames, size_t num_frames)
+{
+    for (size_t i = 0; i < num_frames; ++i) {
+        free(frames[i].registers);
+    }
+    free(frames);
+}
+
+uint64_t get_total_stack_size(call_stack_state_t *state)
+{
+    uint64_t total_size = 0;
+    for (size_t i = 0; i + 1 < state->depth; ++i) {
+        // Add 8 to account for the return address.
+        total_size += state->frames[i].size + 8;
```

Comment:
There are lots of `8`s in the source code. I do wonder if we should have used a macros to define these and attach semantic meaning to them.